Snakefile

```diff
@@ -100,9 +100,10 @@ def aggregate_input(wildcards):
     """
     with open(checkpoints.mapped_reads.get(sample=wildcards.sample, reference=wildcards.reference).output[0]) as f:
         summary = f.readlines()
-        min_reads = float(summary[0].split()[-1])
-        mapped = float(summary[1].split()[-1])
-        if mapped <= min_reads:
+        all_segments_aligned = eval(summary[0].split()[-1])
```

Comment:
The use of `eval` makes me suspect about other ways of doing this. After a quick skim, is what's happening here a boolean is getting serialized to the text file as a string and you want to get it back to a boolean?

I think a better approach would be to serialize the data you want (including the floats below) to a JSON document.

Reply:
Yup, that's exactly what's happening. Definitely better to dump to a JSON document!